### PR TITLE
Change Helm Directory Structure

### PR DESCRIPTION
**Pull request by goruha**
## what
* make it more closely resemble http://github.com/kubernetes/charts

## why
* easier to deploy to object stores like S3 and to bundle in docker images (see: https://github.com/cloudposse/charts/pull/42)

## who
@cloudposse/engineering

### CODE REVIEW

File: modules/helm/Makefile.repo [hunk 1]
```diff
@@ -59,25 +59,26 @@ helm\:repo\:fix-perms:
...
 
 .PHONY : helm\:repo\:lint
-## Lint library charts
+## Lint charts
 helm\:repo\:lint:
 	$(call assert-set,REPO_PATH)
-	@echo "## Linting charts $(shell pwd)/library"
-	@find ./$(REPO_PATH)/library -maxdepth 1 -mindepth 1 -type d | \
+	@echo "## Linting charts $(shell pwd)"
+	@find ./$(REPO_PATH) -maxdepth 1 -mindepth 1 -type d | \
```
Comment (osterman) [inline modules/helm/Makefile.repo:81]: I don't think this should be prefixed with `./`; that is what `REPO_PATH` is for

File: modules/helm/Makefile.repo [hunk 2]
```diff
@@ -86,8 +87,8 @@ helm\:repo\:lint:
 helm\:repo\:clean:
 	$(call assert-set,REPO_PATH)
 	@rm -f *.tgz
-	@find ./$(REPO_PATH)/library -maxdepth 3 -mindepth 3 -type f -name '*.tgz' -delete
-	@rm -f ./$(REPO_PATH)/packages/index.yaml ./packages/*.tgz
+	@find ./$(REPO_PATH) -maxdepth 3 -mindepth 3 -type f -name '*.tgz' -delete
+	@rm -f ./packages/$(REPO_PATH)/index.yaml ./packages/*.tgz
```
Comment (osterman) [inline modules/helm/Makefile.repo:91]: We need to define 3 new ENVs
`REPO_NAME ?= incubator # for example`
`REPO_PATH ?= ./$(REPO_NAME)`
`PACKAGE_PATH ?= ./packages/$(REPO_NAME)`

File: modules/helm/Makefile.repo (hunk 1 above)
Comment (goruha) [reply modules/helm/Makefile.repo:81]: Done

File: modules/helm/Makefile.repo (hunk 2 above)
Comment (goruha) [reply modules/helm/Makefile.repo:91]: Done

**Pull request merged**